### ISSUE TITLE
test_sslverify uses deprecated function log.flushErrors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2766]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#2766|
|Type|defect|
|Created|2007-08-02 10:16:34Z|

We should remove these calls. It also needs a bit of cleanups, like removing tabs in it.

<details><summary>Searchable metadata</summary>

```
trac-id__2766 2766
type__defect defect
reporter__therve therve
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1186049794000000 1186049794000000
changetime__1194453331000000 1194453331000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [20916])  * Remove calls to flushErrors
 * 80 cols
 * Remove tabs
 * Remove trailing whitespaces
 * Support lack of pyOpenSSL

Refs [#2766](https://github.com/twisted/twisted/issues/2766)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

Work is done in est-sslverify-cleanups-2766, will push for review later.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

Branch is test-sslverify-cleanups-2766

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @therve|
|-|-|

Man, tabs.

Some tests changed without getting docstrings:

  * testAbbreviatingDistinguishedNames
  * testAllowedAnonymousClientConnection
  * testRefusedAnonymousClientConnection
  * testFailedCertificateVerification
  * testSuccessfulCertificateVerification
  * testSuccessfulSymmetricSelfSignedCertificateVerification
  * testVerification

Otherwise generally looks good.  It might be nice to define the skips inside the test case definitions instead of adding them to the end.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve commented|
|-|-|

(In [21295]) Add some docstrings

Refs [#2766](https://github.com/twisted/twisted/issues/2766)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve removed owner|
|-|-|

Back to review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @therve|
|-|-|

OpenSSLOptions.skip is set in two places now, and Constructor.skip isn't set anywhere.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve set owner to @exarkun|
|-|-|

Damn, sorry, I replaced it. Is it ok to merge now ?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set owner to @therve|
|-|-|

Looks good

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_12"></a>|@therve set status to `closed`|
|-|-|

(In [21389]) Merge test-sslverify-cleanups-2766

Author: therve
Reviewer: exarkun
Fixes [#2766](https://github.com/twisted/twisted/issues/2766)

Make some cleanups in twisted.test.test_sslverify:
 * Replace call to flushErrors
 * Remove tabs, trailing whitespaces
 * Support lack of PyOpenSSL
 * Add some docstrings

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_13"></a>|@therve commented|
|-|-|

(In [21390]) Revert r21389: regressions under win32

Refs [#2766](https://github.com/twisted/twisted/issues/2766)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_14"></a>|@therve set status to `reopened`|
|-|-|

There is an error on the win32 slaves:

```
[ERROR]: twisted.test.test_sslverify.OpenSSLOptions.test_refusedAnonymousClientConnection

Traceback (most recent call last):
Failure: twisted.internet.error.ConnectionLost: Connection to the other side was lost in a non-clean fashion.
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_15"></a>|@therve removed owner<br>@therve set status to `new`|
|-|-|

That should be better now.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_16"></a>|@radix set owner to @therve|
|-|-|

I'm unhappy with the stuff in setUp. What do you think of storing those keys in strings or on the filesystem? I'm happy with your fixes for the win32 test failures.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_17"></a>|@therve commented|
|-|-|

Replying to [radix](#note_16):
> I'm unhappy with the stuff in setUp. What do you think of storing those keys in strings or on the filesystem?

I don't think it brings something. I see 2 options:
 * generate keys of size 512 instead of 1024, it's much faster
 * keep on generating keys of 1024 for each test, it's just a little bit more slower.

To give an idea, with 512 the tests take roughly 2 secs, and 5 with 1024 (1.2 with current version).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_18"></a>|@therve commented|
|-|-|

(In [21621]) Use keys of length 512.

Refs [#2766](https://github.com/twisted/twisted/issues/2766)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_19"></a>|@therve removed owner|
|-|-|

I went on and did the first solution. Back to review (in **test-sslverify-cleanups-2766**).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_20"></a>|@exarkun set owner to @therve|
|-|-|

Unshared state is great.  The slowdown is okay for now I think.  I kind of like the idea of keeping certs for the test on disk, but that's a separate issue.

Tests pass on Windows now, it seems, and still work on Linux.  The Win32 ConnectionLost behavior is a bug we should fix, I suppose (although man, screw Windows).  Please file a ticket for that and reference it in the comment above ```cResult.trap```.

Do that, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_21"></a>|@therve commented|
|-|-|

(In [21629]) Mention the ticket in the comment

Refs [#2766](https://github.com/twisted/twisted/issues/2766)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_22"></a>|@therve set status to `closed`|
|-|-|

(In [21630]) Merge test-sslverify-cleanups-2766

Author: therve
Reviewer: exarkun
Fixes [#2766](https://github.com/twisted/twisted/issues/2766)

Make some cleanups in twisted.test.test_sslverify:
 * Replace call to flushErrors
 * Remove tabs, trailing whitespaces
 * Support lack of PyOpenSSL
 * Add some docstrings

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
